File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/views/utils/display_temperature_unit/DisplayTemperatureUnitStrategy.java
package coolpharaoh.tee.speicher.tea.timer.views.utils.display_temperature_unit;

import static java.lang.String.valueOf;

public interface DisplayTemperatureUnitStrategy {

    String getTextIdShowTea(int temperature);

    String getTextNewTea(int temperature);

    static String getFormattedTemperature(final int temperature) {
        String text = "-";
        if (exist(temperature)) {
            text = valueOf(temperature);
        }
        return text;
    }

    static boolean exist(final int temperature) {
        return temperature != -500;
    }
}

File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/views/utils/display_temperature_unit/DisplayTemperatureUnitStrategy.kt
package coolpharaoh.tee.speicher.tea.timer.views.utils.display_temperature_unit

interface DisplayTemperatureUnitStrategy {

    fun getTextIdShowTea(temperature: Int): String

    fun getTextNewTea(temperature: Int): String

    companion object {
        @JvmStatic
        fun getFormattedTemperature(temperature: Int): String {
            var text = "-"
            if (exist(temperature)) {
                text = temperature.toString()
            }
            return text
        }

        fun exist(temperature: Int): Boolean {
            return temperature != -500
        }
    }
}